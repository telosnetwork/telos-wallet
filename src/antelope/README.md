## Antelope library

Antelope is a TypeScript library that streamlines the process of developing blockchain applications for the Antelope Alliance, which includes platforms like EOS, Telos, WAX, UX and their respectives EVMs. The library enables MetaMask-based authentication for EVM blockchains as long as any UAL-compatible wallet for native Antelope blockchains. Utilizing Pinia as a storage solution, Antelope maintains and stores application data, and leverages observables from RxJS to trigger internal events, to which developers can subscribe and respond.

It is essential to note that Antelope is still a work in progress, and its architecture is subject to design changes. This library incorporates the experiences of multiple developers and is built from code fragments currently in use by websites under the Telos Network group. The objective is to aggregate all this code into a singular library and subsequently adapt the websites to implement this new library.


## External dependencies
```typescript
import { App } from 'vue';
import { Authenticator, RpcEndpoint } from "universal-authenticator-library";
import { Subject, BehaviorSubject, filter } from 'rxjs';
import { StoreCallback } from '@quasar/app-webpack';
import { store } from 'quasar/wrappers';
import { createPinia, defineStore } from 'pinia';
import { ethers } from 'ethers';
import detectEthereumProvider from '@metamask/detect-provider';
import { ExternalProvider } from '@ethersproject/providers';
```