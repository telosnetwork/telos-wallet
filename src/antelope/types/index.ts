// interfaces for antelope
export * from 'src/antelope/types/ABIv1';
export * from 'src/antelope/types/Actions';
export * from 'src/antelope/types/Api';
export * from 'src/antelope/types/ChainInfo';
export * from 'src/antelope/types/ChainSettings';
export * from 'src/antelope/types/EvmContractData';
export * from 'src/antelope/types/EvmLog';
export * from 'src/antelope/types/EvmBlockData';
export * from 'src/antelope/types/EvmTransaction';
export * from 'src/antelope/types/ExceptionError';
export * from 'src/antelope/types/Filters';
export * from 'src/antelope/types/PriceData';
export * from 'src/antelope/types/Proposals';
export * from 'src/antelope/types/Producers';
export * from 'src/antelope/types/KeyAccounts';
export * from 'src/antelope/types/Label';
export * from 'src/antelope/types/Theme';
export * from 'src/antelope/types/TransactionV1';


// classes for antelope
export * from 'src/antelope/types/AntelopeError';

// interfaces for antelope evm-abi
export * from 'src/antelope/stores/utils/abi';
