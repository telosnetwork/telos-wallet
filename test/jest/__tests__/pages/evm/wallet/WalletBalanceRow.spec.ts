import { shallowMount } from '@vue/test-utils';

const fakeBuyMoreLink = 'fake';
const fakeStlosContractAddress = '0x'.concat('9'.repeat(40));
const fakeWtlosContractAddress = '0x'.concat('8'.repeat(40));
const fakeTokenContractAddress = '0x'.concat('7'.repeat(40));
const storeMock = {
    useChainStore: () => ({
        currentChain: {
            settings: {
                getBuyMoreOfTokenLink: () => fakeBuyMoreLink,
                getStakedNativeTokenAddress: () => fakeStlosContractAddress,
                getWrappedNativeTokenAddress: () => fakeWtlosContractAddress,
                getExplorerUrl: () => 'fake-url',
            },
        },
    }),
    useUserStore: () => ({
        fiatLocale: 'en-US',
        fiatCurrency: 'USD',
    }),
    useEVMStore: () => ({
        isMetamaskSupported: false,
    }),
};

import WalletBalanceRow from 'pages/evm/wallet/WalletBalanceRow.vue';
import { EvmToken, NativeCurrencyAddress } from 'src/antelope/types';

import { stubWithSlot } from 'test/jest/testing-helpers';

jest.mock('src/antelope', () => storeMock);

describe('WalletBalanceRow.vue', () => {
    const stubs = {
        'ToolTip': stubWithSlot('ToolTip'),
        'q-menu': stubWithSlot('q-menu'),
        'q-btn': stubWithSlot('q-btn'),
    };

    const directives = {
        'close-popup': ()  => '',
    };

    const qScreenMock = (isMobile: boolean) => ({
        $q: {
            screen: {
                lt: {
                    sm: isMobile,
                },
            },
        },
    });

    it('should have the correct name', () => {
        expect(WalletBalanceRow.name).toBe('WalletBalanceRow');
    });

    it('should render correctly on desktop devices', () => {
        const wrapper = shallowMount(WalletBalanceRow, {
            global: {
                stubs,
                directives,
                mocks: {
                    ...qScreenMock(false),
                },
            },
            props: {
                token: {
                    address: fakeStlosContractAddress,
                    symbol: 'STLOS',
                    name: 'Staked TLOS',
                    logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                    decimals: 18,
                    balance: '3642.0243',
                    fullBalance: '3642.024318091460206147',
                } as EvmToken,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render correctly on mobile devices', () => {
        const wrapper = shallowMount(WalletBalanceRow, {
            global: {
                stubs,
                directives,
                mocks: {
                    ...qScreenMock(true),
                },
            },
            props: {
                token: {
                    address: fakeStlosContractAddress,
                    symbol: 'STLOS',
                    name: 'Staked TLOS',
                    logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                    decimals: 18,
                    balance: '3642.0243',
                    fullBalance: '3642.024318091460206147',
                } as EvmToken,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    describe('should render the right overflow items when the token is', () => {
        const routerMock = {
            push: jest.fn(),
        };

        const originalWindowOpen = window.open;

        beforeAll(() => {
            window.open = jest.fn();
        });

        beforeEach(() => {
            jest.clearAllMocks();
        });

        afterAll(() => {
            window.open = originalWindowOpen;
        });

        test('TLOS', async () => {
            const wrapper = shallowMount(WalletBalanceRow, {
                global: {
                    stubs,
                    directives,
                    mocks: {
                        $router: routerMock,
                        ...qScreenMock(true),
                    },
                },
                props: {
                    token: {
                        symbol: 'TLOS',
                        name: 'Telos',
                        logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                        decimals: 18,
                        balance: '3642.0243',
                        fullBalance: '3642.024318091460206147',
                        address: NativeCurrencyAddress,
                    } as EvmToken,
                },
            });

            // snapshot contains c-wallet-balance-row__overflow-li for stake, buy, wrap, send
            expect(wrapper.element).toMatchSnapshot();

            const [
                // stakeLink,
                buyLink,
                // wrapLink,
                sendLink,
            ] = wrapper.findAll('.c-wallet-balance-row__overflow-li');

            // await stakeLink.trigger('click');
            await buyLink.trigger('click');
            // await wrapLink.trigger('click');
            await sendLink.trigger('click');

            expect(window.open).toHaveBeenCalledTimes(1);
            // expect(routerMock.push).toHaveBeenCalledTimes(3);
            expect(routerMock.push).toHaveBeenCalledTimes(1);

            // expect(routerMock.push.mock.calls[0][0]).toEqual(expect.objectContaining({
            //     name: 'evm-staking',
            // }));

            // expect(routerMock.push.mock.calls[1][0]).toEqual(expect.objectContaining({
            //     name: 'evm-wrap',
            // }));

            expect(routerMock.push.mock.calls[0][0]).toEqual(expect.objectContaining({
                name: 'evm-send',
                query: { },
            }));

            expect(window.open).toHaveBeenCalledWith(fakeBuyMoreLink, '_blank');
        });

        test('STLOS', async () => {
            const wrapper = shallowMount(WalletBalanceRow, {
                global: {
                    stubs,
                    directives,
                    mocks: {
                        $router: routerMock,
                        ...qScreenMock(true),
                    },
                },
                props: {
                    token: {
                        address: fakeStlosContractAddress,
                        symbol: 'STLOS',
                        name: 'Staked TLOS',
                        logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                        decimals: 18,
                        balance: '3642.0243',
                        fullBalance: '3642.024318091460206147',
                    } as EvmToken,
                },
            });

            // snapshot contains c-wallet-balance-row__overflow-li for unstake, contract, send
            expect(wrapper.element).toMatchSnapshot();

            const [
                // stakeLink,
                contractLink,
                sendLink,
            ] = wrapper.findAll('.c-wallet-balance-row__overflow-li');

            // await stakeLink.trigger('click');
            await contractLink.trigger('click');
            await sendLink.trigger('click');

            expect(window.open).toHaveBeenCalledTimes(1);
            // expect(routerMock.push).toHaveBeenCalledTimes(2);
            expect(routerMock.push).toHaveBeenCalledTimes(1);

            // expect(routerMock.push.mock.calls[0][0]).toEqual(expect.objectContaining({
            //     name: 'evm-staking',
            // }));

            expect(routerMock.push.mock.calls[0][0]).toEqual(expect.objectContaining({
                name: 'evm-send',
                query: {
                    token: fakeStlosContractAddress,
                },
            }));

            expect(window.open).toHaveBeenCalledWith(
                `fake-url/address/${fakeStlosContractAddress}`,
                '_blank',
            );
        });

        test('WTLOS', async () => {
            const wrapper = shallowMount(WalletBalanceRow, {
                global: {
                    stubs,
                    directives,
                    mocks: {
                        $router: routerMock,
                        ...qScreenMock(true),
                    },
                },
                props: {
                    token: {
                        address: fakeWtlosContractAddress,
                        symbol: 'WTLOS',
                        name: 'Wrapped TLOS',
                        logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                        decimals: 18,
                        balance: '3642.0243',
                        fullBalance: '3642.024318091460206147',
                    } as EvmToken,
                },
            });

            // snapshot contains c-wallet-balance-row__overflow-li for unwrap, contract, send
            expect(wrapper.element).toMatchSnapshot();

            const [
                // unwrapLink,
                contractLink,
                sendLink,
            ] = wrapper.findAll('.c-wallet-balance-row__overflow-li');

            // await unwrapLink.trigger('click');
            await contractLink.trigger('click');
            await sendLink.trigger('click');

            expect(window.open).toHaveBeenCalledTimes(1);
            // expect(routerMock.push).toHaveBeenCalledTimes(2);
            expect(routerMock.push).toHaveBeenCalledTimes(1);

            // expect(routerMock.push.mock.calls[0][0]).toEqual(expect.objectContaining({
            //     name: 'evm-wrap',
            //     query: { tab: 'unwrap' },
            // }));

            expect(routerMock.push.mock.calls[0][0]).toEqual(expect.objectContaining({
                name: 'evm-send',
                query: {
                    token: fakeWtlosContractAddress,
                },
            }));

            expect(window.open).toHaveBeenCalledWith(
                `fake-url/address/${fakeWtlosContractAddress}`,
                '_blank',
            );
        });

        test('not a system token', async () => {
            const wrapper = shallowMount(WalletBalanceRow, {
                global: {
                    stubs,
                    directives,
                    mocks: {
                        $router: routerMock,
                        ...qScreenMock(true),
                    },
                },
                props: {
                    token: {
                        address: fakeTokenContractAddress,
                        symbol: 'SHIB',
                        name: 'Shiba',
                        logoURI: 'https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png',
                        decimals: 18,
                        balance: '3642.0243',
                        fullBalance: '3642.024318091460206147',
                    } as EvmToken,
                },
            });

            // snapshot contains c-wallet-balance-row__overflow-li for contract, send
            expect(wrapper.element).toMatchSnapshot();

            const [
                contractLink,
                sendLink,
            ] = wrapper.findAll('.c-wallet-balance-row__overflow-li');

            await contractLink.trigger('click');
            await sendLink.trigger('click');

            expect(window.open).toHaveBeenCalledTimes(1);
            expect(routerMock.push).toHaveBeenCalledTimes(1);

            expect(routerMock.push.mock.calls[0][0]).toEqual(expect.objectContaining({
                name: 'evm-send',
                query: {
                    token: fakeTokenContractAddress,
                },
            }));

            expect(window.open).toHaveBeenCalledWith(
                `fake-url/address/${fakeTokenContractAddress}`,
                '_blank',
            );
        });
    });
});
