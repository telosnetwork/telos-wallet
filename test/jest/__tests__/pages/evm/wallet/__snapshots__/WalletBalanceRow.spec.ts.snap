// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletBalanceRow.vue should render correctly on desktop devices (Staked TLOS 1) 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <span
        class="c-wallet-balance-row__token_name"
      >
        Staked TLOS
      </span>
      <br />
      <span
        class="o-text--small"
      >
        $0.1900
      </span>
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        
        $691.98
        
        <!--v-if-->
      </div>
      <span
        class="c-wallet-balance-row__secondary-amount"
      >
        
        3,642.0243 STLOS
        
      </span>
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              evm_wallet.send
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon c-wallet-balance-row__overflow-icon--stroke"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.add_to_metamask
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render correctly on mobile devices (Staked TLOS 2) 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <span
        class="c-wallet-balance-row__token_name"
      >
        Staked TLOS
      </span>
      <br />
      <span
        class="o-text--small"
      >
        $0.1900
      </span>
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        
        $691.98
        
        <!--v-if-->
      </div>
      <span
        class="c-wallet-balance-row__secondary-amount"
      >
        <tooltip-stub
          hide-icon="true"
          text="3,642.0243 STLOS"
        >
          
          3.64K STLOS
          
        </tooltip-stub>
      </span>
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              evm_wallet.send
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon c-wallet-balance-row__overflow-icon--stroke"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.add_to_metamask
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render the right overflow items when the token is STLOS 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <span
        class="c-wallet-balance-row__token_name"
      >
        Staked TLOS
      </span>
      <br />
      <span
        class="o-text--small"
      />
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        <tooltip-stub
          hide-icon="true"
          text="3,642.0243 STLOS"
        >
          
          3.64K
          
        </tooltip-stub>
        <tooltip-stub
          icon="info"
          warnings="evm_wallet.no_fiat_value"
        >
          
          
        </tooltip-stub>
      </div>
      <!--v-if-->
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              evm_wallet.send
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon c-wallet-balance-row__overflow-icon--stroke"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.add_to_metamask
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render the right overflow items when the token is TLOS 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <span
        class="c-wallet-balance-row__token_name"
      >
        Telos
      </span>
      <br />
      <span
        class="o-text--small"
      >
        $0.1900
      </span>
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        
        $691.98
        
        <!--v-if-->
      </div>
      <span
        class="c-wallet-balance-row__secondary-amount"
      >
        <tooltip-stub
          hide-icon="true"
          text="3,642.0243 TLOS"
        >
          
          3.64K TLOS
          
        </tooltip-stub>
      </span>
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              evm_wallet.buy
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              evm_wallet.send
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.wrap
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render the right overflow items when the token is WTLOS 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <span
        class="c-wallet-balance-row__token_name"
      >
        Wrapped TLOS
      </span>
      <br />
      <span
        class="o-text--small"
      >
        $0.1900
      </span>
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        
        $691.98
        
        <!--v-if-->
      </div>
      <span
        class="c-wallet-balance-row__secondary-amount"
      >
        <tooltip-stub
          hide-icon="true"
          text="3,642.0243 WTLOS"
        >
          
          3.64K WTLOS
          
        </tooltip-stub>
      </span>
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              evm_wallet.send
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.unwrap
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-3"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon c-wallet-balance-row__overflow-icon--stroke"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-3"
            >
              evm_wallet.add_to_metamask
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;

exports[`WalletBalanceRow.vue should render the right overflow items when the token is not a system token 1`] = `
<div
  class="c-wallet-balance-row"
>
  <div
    class="c-wallet-balance-row__left-container"
  >
    <img
      alt="evm_wallet.token_logo_alt"
      aria-hidden="true"
      class="c-wallet-balance-row__logo"
      height="40"
      src="https://raw.githubusercontent.com/telosnetwork/teloscan/master/public/stlos-logo.png"
      width="40"
    />
    <div>
      <span
        class="c-wallet-balance-row__token_name"
      >
        Shiba
      </span>
      <br />
      <span
        class="o-text--small"
      />
    </div>
  </div>
  <div
    class="c-wallet-balance-row__right-container"
  >
    <div
      class="c-wallet-balance-row__balance-container"
    >
      <div
        class="c-wallet-balance-row__primary-amount"
      >
        <tooltip-stub
          hide-icon="true"
          text="3,642.0243 SHIB"
        >
          
          3.64K
          
        </tooltip-stub>
        <tooltip-stub
          icon="info"
          warnings="evm_wallet.no_fiat_value"
        >
          
          
        </tooltip-stub>
      </div>
      <!--v-if-->
    </div>
    <q-btn-stub
      aria-label="evm_wallet.balance_row_actions_aria"
      class="c-wallet-balance-row__overflow"
      dense=""
      flat=""
      icon="more_vert"
      no-icon-animation=""
    >
      
      <q-menu-stub
        anchor="bottom end"
        offset="0,16"
        self="top right"
      >
        
        <ul
          class="c-wallet-balance-row__overflow-ul"
        >
          
          <li
            aria-labelledby="overflow-text-0"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-0"
            >
              global.contract
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-1"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-1"
            >
              evm_wallet.send
            </span>
          </li>
          <li
            aria-labelledby="overflow-text-2"
            class="c-wallet-balance-row__overflow-li"
            tabindex="0"
          >
            <div
              class="c-wallet-balance-row__overflow-icon-wrapper"
            >
              <inline-svg-stub
                aria-hidden="true"
                class="c-wallet-balance-row__overflow-icon c-wallet-balance-row__overflow-icon--stroke"
                keepduringloading="true"
                src=""
                transformsource="[Function]"
              />
            </div>
            <span
              class="c-wallet-balance-row__overflow-text"
              id="overflow-text-2"
            >
              evm_wallet.add_to_metamask
            </span>
          </li>
          
        </ul>
        
      </q-menu-stub>
      
    </q-btn-stub>
  </div>
</div>
`;
